### PR TITLE
Feature/cdap common deprecations

### PR DESCRIPTION
**Pull request by gsps1**
remove deprecations in cdap-common. most of them are straightforward removal, except the security announce address, which didn't follow the pattern we usually follow for deprecations.

### CODE REVIEW

**Comment by gsps1** [comment]
build running : https://builds.cask.co/browse/CDAP-DUT6189

File: cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/AuthenticationHandler.java [hunk 1]
```diff
@@ -296,24 +296,6 @@ private static Pattern createBypassPattern(CConfiguration cConf) {
     if (!urls.isEmpty()) {
```
Comment (ecfm) [inline cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/AuthenticationHandler.java]: Can remove

**Review by ecfm — COMMENTED** [review]
One comment, otherwise LGTM

**Comment by gsps1** [comment]
@maochf take a look at it again. thanks!

**Comment by ecfm** [comment]
LGTM

**Comment by gsps1** [comment]
build green, merging.

**Pull request merged**